File: public/docs/_examples/tutorial/ts/src/app/dashboard.component.ts
import {Component, OnInit} from 'angular2/angular2';
import {Router} from 'angular2/router';
import {Hero} from './hero';
import {HeroService} from './hero.service';
import {ROUTE_NAMES} from './routes';

@Component({
  selector: 'my-dashboard',
  templateUrl: 'app/dashboard.component.html',
  styleUrls: ['app/dashboard.component.css']
})
export class DashboardComponent implements OnInit {
  public heroes: Hero[] = [];

  constructor(private _heroService: HeroService, private _router: Router) { }

  ngOnInit() {
    this._heroService.getHeroes().then(heroes => this.heroes = heroes);
  }

  gotoDetail(hero: Hero) {
    this._router.navigate([ROUTE_NAMES.heroDetail, { id: hero.id }]);
  }
}

File: public/docs/_examples/tutorial/dart/lib/dashboard_component.dart
import 'package:angular2/core.dart';
import 'package:angular2/router.dart';

import 'hero.dart';
import 'hero_service.dart';

@Component(
    selector: 'my-dashboard',
    templateUrl: 'dashboard_component.html',
    styleUrls: const ['dashboard_component.css'])
class DashboardComponent implements OnInit {
  List<Hero> heroes = [];

  final HeroService _heroService;
  final Router _router;
  DashboardComponent(this._heroService, this._router);

  ngOnInit() async {
    _heroService
        .getHeroes()
        .then((heroes) => this.heroes = heroes.sublist(1, 5));
  }

  gotoDetail(Hero hero) {
    _router.navigate([
      'HeroDetail',
      {'id': hero.id}
    ]);
  }
}

